### PR TITLE
feat: stake function

### PR DESCRIPTION
**Pull request by 0xsambugs**
We are now adding a function that allows the owner to stake a certain amount of tokens

### CODE REVIEW

File: src/euler/staking/StakeableEulerERC4626.sol [hunk 1]
```diff
@@ -67,6 +67,12 @@ contract StakeableEulerERC4626 is EulerERC4626, Owned {
         if (stakingRewards_.stakingToken() != address(eToken)) revert StakeableEulerERC4626__InvalidRewardContract();
 
         stakingRewards = stakingRewards_;
+        ERC20(address(eToken)).safeApprove(address(stakingRewards_), type(uint256).max);
```
Comment (0xged) [inline src/euler/staking/StakeableEulerERC4626.sol:70]: 🧠

File: src/test/euler/staking/StakeableEulerERC4626.t.sol [hunk 2]
```diff
@@ -69,6 +72,34 @@ contract StakeableEulerERC4626Test is Test {
...
+
+    function testStaking() public {
+        _setStakingContract();
+        uint256 deposited = 10000;
+        uint256 staked = 1000;
+
+        vm.prank(alice);
+        underlying.approve(address(vault), deposited);
+        vm.prank(alice);
+        vault.deposit(deposited, alice);
```
Comment (0xged) [inline src/test/euler/staking/StakeableEulerERC4626.t.sol:90]: Do you need to prank alice two times? Is Foundry that shitty? 🤔

**Review by 0xged — APPROVED** [review]
gj!

File: src/test/euler/staking/StakeableEulerERC4626.t.sol (hunk 2 above)
Comment (0xsambugs) [reply src/test/euler/staking/StakeableEulerERC4626.t.sol:90]: `prank` sets the `msg.sender` for the next call

**Pull request merged**